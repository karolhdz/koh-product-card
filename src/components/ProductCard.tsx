
import styles from '../styles/styles.module.css'
import { useProduct } from '../hooks/useProduct';
import  React from 'react';
import { Provider } from '../context/ProductContext';
import { InitialValue, onChangeArgs, Product, ProductCardHandlers } from '../interfaces/ProductInterfaces';

export interface Props {
    product: Product,
    // children?: ReactElement | ReactElement[]
    children: ( args: ProductCardHandlers ) => JSX.Element
    className?: string,
    style?: React.CSSProperties,
    onChange?: (args: onChangeArgs) => void,
    value?: number,
    initialValues?: InitialValue
}

export const ProductCard = ({ children, product, className, style, onChange, value, initialValues}: Props) => {



    const { counter, onHandleIncrease, maxCount, isMaxCountReached, reset } = useProduct({ onChange, product, value, initialValues});

    return (
        <Provider value={{ counter, onHandleIncrease, product, maxCount }}>
            <div 
                className={`${styles.productCard} ${className}`}
                style={style}
            >

                {children({
                    count: counter,
                    isMaxCountReached,
                    maxCount: initialValues?.maxCount,
                    onHandleIncrease,
                    reset,
                    product,
                })}
                

            </div>
        </Provider>

    )
}

