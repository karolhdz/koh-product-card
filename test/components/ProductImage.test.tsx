import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/productos';

describe('Pruebas sobre el componente <ProductImage/>', () => {
    test('Debe de mostrar el componente correctamente con la imagen personalizado', () => {

        const wrapper = renderer.create(
            <ProductImage img='cosa1.jpg' />
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
      
    });

    test('Debe de mostar el componente con la imagen  del producto', () => {

        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                       <ProductImage/> 
                    )
                }
            </ProductCard>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
      
    })
    
})
