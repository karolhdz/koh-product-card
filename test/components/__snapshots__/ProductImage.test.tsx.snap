// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <ProductImage/> Debe de mostar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffe-mug.jpg"
  />
</div>
`;

exports[`Pruebas sobre el componente <ProductImage/> Debe de mostar el componente con la imagen  del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffe-mug.jpg"
  />
</div>
`;

exports[`Pruebas sobre el componente <ProductImage/> Debe de mostrar el componente correctamente con la imagen personalizado 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="cosa1.jpg"
/>
`;
